usr/sbin/p11sak/p11sak.c

```diff
@@ -2550,6 +2775,49 @@ static CK_RV close_session(CK_SESSION_HANDLE session)
     return rc;
 }
 
+static CK_RV parse_file() {
+    CK_RV rc = CKR_OK;
+    FILE *fp;
+    char *file_loc = getenv("P11SAK_DEFAULT_CONF_FILE");
+
+    // open and parse file
+    if ((fp = fopen(file_loc, "r")) == NULL) {
+        if (file_loc) {
+            fprintf(stderr, "Failed to load %s: %s\n", file_loc, strerror(errno));
```

Comment:
If the environment variable P11SAK_DEFAULT_CONF_FILE is specified by the user (i.e. file_loc not not NULL here), then I would just return an error if the file can not be opened and stop. I would not try to use the other file locations in this case. This could hide an erroneously set environment variable.

I would only try the user-specific config file and then the global config file, of the environment variable is NOT specified. 

I.e. something like this.

```
char *file_loc = getenv("P11SAK_DEFAULT_CONF_FILE");

if (file_loc != NULL) {
    if ((fp = fopen(file_loc, "r")) == NULL) {
    .....
} else {
   pw = getpwuid(getuid());
   ....
   if (!fp) {
            file_loc = P11SAK_DEFAULT_CONF_FILE;
            if ((fp = fopen(file_loc, "r")) == NULL)
            ....
    }
}
```

Reply:
Done